extensions/amp-app-banner/1.0/component/component.js

```diff
@@ -0,0 +1,143 @@
+ * @param {BentoAppBannerDef.RawProps} props
+ * @return {JSX.Element}
+ * @constructor
+ */
+export function AppBanner({
+  children,
+  dismissButtonAriaLabel = 'Dismiss',
+  onDismiss,
+  onInstall,
+  ...rest
+}) {
+  const styles = useStyles();
+
+  const bannerRef = useRef(null);
+
+  useEffect(() => {
+    // We expect the children to include an open-button, like: <button open-button>Install App</button>
+    let button = bannerRef.current.querySelector('button[open-button]');
+
+    // When used as a web-component, we must search the slot for the open-button:
+    if (!button) {
+      const slot = bannerRef.current.querySelector('slot');
+      if (slot) {
+        button = querySelectorInSlot(slot, 'button[open-button]');
+      }
+    }
+
+    if (!button) {
+      logger.error(
+        'BENTO-APP-BANNER',
+        'Component children should contain a <button open-button>'
+      );
+      return;
+    }
+
+    button.addEventListener('click', onInstall);
+    return () => button.removeEventListener('click', onInstall);
+  }, [onInstall]);
+
+  return (
+    <ContainWrapper {...rest}>
+      <div class={styles.banner} ref={bannerRef}>
+        <div class={styles.bannerPadding} />
+        <button
+          class={styles.dismiss}
+          aria-label={dismissButtonAriaLabel}
+          onClick={onDismiss}
+        />
+        {children}
+      </div>
+    </ContainWrapper>
+  );
+}
+
+/**
+ * @param {BentoAppBannerDef.Props} props
+ * @return {JSX.Element|null}
+ * @constructor
+ */
+function AppBannerIOS(props) {
+  const appInfo = useMemo(getIOSAppInfo, []);
+  if (!appInfo) {
+    return null;
+  }
+
+  return <AppBanner {...props} onInstall={appInfo.openOrInstall} />;
+}
+
+/**
+ * @param {BentoAppBannerDef.Props} props
+ * @return {JSX.Element|null}
+ * @constructor
+ */
+function AppBannerAndroid(props) {
+  const appInfo = useMemo(getAndroidAppInfo, []);
+  if (!appInfo) {
+    return null;
+  }
+
+  return <AppBanner {...props} onInstall={appInfo.openOrInstall} />;
+}
+
+/**
+ * @param {!BentoAppBanner.Props} props
+ * @return {PreactDef.Renderable}
+ */
+export function BentoAppBanner(props) {
+  if (!props.id) {
+    logger.warn('BENTO-APP-BANNER', 'component should have an id');
+  }
+  const [isDismissed, setDismissed] = useLocalStorage(
+    getStorageKey(props.id),
+    false
+  );
+  if (isDismissed) {
+    return null;
+  }
+
+  const AppBannerForCurrentPlatform = platformUtils.isIos()
+    ? AppBannerIOS
+    : platformUtils.isAndroid()
+    ? AppBannerAndroid
+    : null;
+
+  if (!AppBannerForCurrentPlatform) {
+    logger.info(
+      'BENTO-APP-BANNER',
+      'Component not rendered:',
+      'OS not supported'
+    );
+    return null;
+  }
+
+  return (
+    <AppBannerForCurrentPlatform
+      {...props}
+      onDismiss={() => setDismissed(true)}
+    />
+  );
+}
+
+/**
+ * @param {string} id
+ * @return {string}
+ */
+function getStorageKey(id) {
+  return 'bento-app-banner:' + id;
+}
```

Comment:
Because this is the entry point for the build, you'll also need to export any other compoenents you might need here. like how the inline-gallery does it. https://github.com/ampproject/amphtml/blob/main/extensions/amp-inline-gallery/1.0/component.js#L32
```suggestion
}

export * from './ios';
export * from './android';
```